src/enigma2/Admin.cpp

```diff
@@ -528,4 +567,116 @@ long long Admin::GetKbFromString(const std::string &stringInMbGbTb) const
   }
 
   return sizeInKb;
+}
+
+PVR_ERROR Admin::GetTunerSignal(PVR_SIGNAL_STATUS &signalStatus)
+{
+  const std::string url = StringUtils::Format("%s%s", Settings::GetInstance().GetConnectionURL().c_str(), "web/tunersignal"); 
+
+  const std::string strXML = WebUtils::GetHttpXML(url);
+  
+  TiXmlDocument xmlDoc;
+  if (!xmlDoc.Parse(strXML.c_str()))
+  {
+    Logger::Log(LEVEL_ERROR, "Unable to parse XML: %s at line %d", xmlDoc.ErrorDesc(), xmlDoc.ErrorRow());
+    return PVR_ERROR_SERVER_ERROR;
+  }
+
+  std::string snrDb;
+  std::string snrPercentage;
+  std::string ber;
+  std::string signalStrength;
+
+  TiXmlHandle hDoc(&xmlDoc);
+  TiXmlElement* pElem;
+  TiXmlHandle hRoot(0);
+
+  pElem = hDoc.FirstChildElement("e2frontendstatus").Element();
```

Comment:
Why not directly `const TiXmlElement* pElem = ` here?

Generally; I would declare vars close to first usage, not at the beginning of the function. We're not doing C here. ;-)

Reply:
Fair, this is just a hang up from original implementation that I copied. Will fix for all in refactoring.